bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableClusterUtilities.java

```diff
@@ -73,6 +70,66 @@ public static BigtableClusterUtilities forAllInstances(String projectId)
   }
 
   /**
+   * @return The instance id associated with the given project, zone and cluster.
+   */
+  public static String lookupInstanceId(String projectId, String clusterId, String zoneId)
+    throws IOException {
+    BigtableClusterUtilities utils;
+    try {
+      utils = BigtableClusterUtilities.forAllInstances(projectId);
+    } catch (GeneralSecurityException e) {
+      throw new RuntimeException("Could not initialize BigtableClusterUtilities", e);
+    }
+
+    try {
+      Cluster cluster = utils.getCluster(clusterId, zoneId);
+      return new BigtableClusterName(cluster.getName()).getInstanceId();
+    } finally {
+      try {
+        utils.close();
+      } catch (Exception e) {
+        logger.warn("Error closing BigtableClusterUtilities: ", e);
+      }
+    }
+  }
+
+  /**
+   * @return The cluster associated with the given project and instance
+   */
+  public static Cluster lookupCluster(String projectId, String instanceId)
```

Comment:
Should lookupSingleCluster throw exception if there are more than one cluster.
Otherwise, this method doesn't make sense.